### ISSUE TITLE
Consultation and some questions about using UniXcoder

### ISSUE THREAD

**Issue by ofirmoshe** [microsoft/CodeBERT#170]
Hi! 
First, I want to commend you for your hard and important work. There is no doubt that you give a leap forward in the field of code analysis and I believe that thanks to you we will see some impressive works in the future.

I have a classification task, where I get a python source code file (.py) and I have to classify it as either related to project A or project B.
I own ~6K labeled python files as my dataset.
Now, I thought about using UniXcoder for extracting an embedding for the python file, then, train another classifier upon these embeddings and the labels I got.
Some questions regards to my idea:

1. What do you believe is the best way to generate a source-code file embedding? Currently, I go over the file in segments of ~3000 characters (which translates to the maximum possible tokens input to UniXcoder), I also take into account the parts where code might get cut in the middle of some logic, so I take the segments with overlaps. Then, I take all the segments' embeddings (using UniXcoder encoder part) and use the mean embedding as my file embedding. Do you believe there's a better way to achieve it?
2. Currently, my input to the model is a tokenized version of my segment code string (as I saw in the readme tutorial). In the paper I saw that you've also used AST format as input, and that might help with more accurate results. Do you think that I better be using the AST format? If so, there's a tutorial and code for this part?
3. Is there any way you recommend to get explanation of the model output (e.g. the input tokens that influenced most)? I would like to be able to somehow interpret the model outputs.

Thank you so much.

**Comment by guoday** [comment]
1. I believe that UniXcoder can generate good segment/statement/function embedding, since I have do a lot of experiments including [AI4Code](https://www.kaggle.com/competitions/AI4Code/leaderboard) competition. The suggestion is that you can try to split a file in to a list of statements or functions, and then use UniXcoder to encode them to a list of embeddings. And then you can fix the embeddings as inputs for a randomly initialized transformer for classification. This's  what I currently do for file-level code task and seems that the solution can achieve relatively good performance.  You can also refer to [this paper](https://arxiv.org/pdf/2208.11271) and I believe it will help you how to model long code.
2. No. I don't suggest you use AST in downstream tasks. We have injected AST knowledge into the model during pre-training part.
3. For this, I suggest you can get attention weights or gradient from the model. You can check the huggingface models API

**Comment by brness** [comment]
Hello, I also encounter the long sentance problem, and my solution is cut the file by max_length, but the prediction seems not so good, any idea of how to split the file into list of statements or functions? @guoday Thx
